cmd/mt-whisper-importer-reader/conversion_test.go

```diff
@@ -0,0 +1,1295 @@
+		{20, 11},
+		{30, 12},
+		{40, 14},
+		{50, 15},
+		{60, 16},
+	}
+
+	expectedResult := map[string][]whisper.Point{
+		"sum": {
+			{30, 33},
+			{60, 45},
+		},
+		"cnt": {
+			{30, 3},
+			{60, 3},
+		},
+	}
+	testDecResolution(t, inData, expectedResult, []string{"sum", "cnt"}, 10, 30)
+}
+
+func TestDecFakeAvgNonFactorResolutions(t *testing.T) {
+	inData := []whisper.Point{
+		{10, 10},
+		{20, 11},
+		{30, 12},
+		{40, 13},
+		{50, 14},
+		{60, 15},
+	}
+
+	expectedResult := map[string][]whisper.Point{
+		"sum": {
+			{15, 10},
+			{30, 23},
+			{45, 13},
+			{60, 29},
+		},
+		"cnt": {
+			{15, 1},
+			{30, 2},
+			{45, 1},
+			{60, 2},
+		},
+	}
+	testDecResolution(t, inData, expectedResult, []string{"sum", "cnt"}, 10, 15)
+}
+
+func TestDecFakeAvgResolutionWithGaps(t *testing.T) {
+	inData := []whisper.Point{
+		{0, 0},
+		{10, 10},
+		{0, 0},
+		{0, 0},
+		{40, 13},
+		{50, 14},
+		{0, 0},
+		{70, 16},
+	}
+
+	expectedResult := map[string][]whisper.Point{
+		"sum": {
+			{20, 10},
+			{40, 13},
+			{60, 14},
+		},
+		"cnt": {
+			{20, 1},
+			{40, 1},
+			{60, 1},
+		},
+	}
+
+	testDecResolution(t, inData, expectedResult, []string{"sum", "cnt"}, 10, 20)
+}
+
+func TestDecResolutionFakeAvgOutOfOrder(t *testing.T) {
+	inData := []whisper.Point{
+		{20, 11},
+		{50, 15},
+		{30, 12},
+		{10, 10},
+		{60, 16},
+		{40, 14},
+	}
+
+	expectedResult := map[string][]whisper.Point{
+		"sum": {
+			{30, 33},
+			{60, 45},
+		},
+		"cnt": {
+			{30, 3},
+			{60, 3},
+		},
+	}
+	testDecResolution(t, inData, expectedResult, []string{"sum", "cnt"}, 10, 30)
+}
+
+func generatePoints(ts, interval uint32, value float64, offset, count int, inc func(float64) float64) []whisper.Point {
+	res := make([]whisper.Point, count)
+	for i := 0; i < count; i++ {
+		res[(i+offset)%count] = whisper.Point{
+			Timestamp: ts,
+			Value:     value,
+		}
+		ts += interval
+		value = inc(value)
+	}
+	return res
+}
+
+func TestRowKeyAgg0(t *testing.T) {
+	res := getRowKey(0, "aaa", "", 0)
+	if res != "aaa" {
+		t.Fatalf("row key for aggregation 0 should equal the id")
+	}
+}
+
+func TestRowKeyAgg1(t *testing.T) {
+	res := getRowKey(1, "aaa", "sum", 60)
+	if res != "aaa_sum_60" {
+		t.Fatalf("row key for aggregation 0 should equal the id")
+	}
+}
+
+func TestEncodedChunksFromPointsWithoutUnfinished(t *testing.T) {
+	points := generatePoints(25200, 10, 10, 0, 8640, func(i float64) float64 { return i + 1 })
+
+	*writeUnfinishedChunks = false
+	chunks := encodedChunksFromPoints(points, 10, 21600)
+
+	if len(chunks) != 4 {
+		t.Fatalf("Expected to get 4 chunks, but got %d", len(chunks))
+	}
+
+	i := 0
+	for _, c := range chunks {
+		iter, err := c.Get()
+		if err != nil {
+			t.Fatalf("Error getting iterator: %s", err)
+		}
+		for iter.Next() {
+			ts, val := iter.Values()
+			if points[i].Timestamp != ts || points[i].Value != val {
+				t.Fatalf("Unexpected value at index %d: %d:%f instead of %d:%f", i, ts, val, points[i].Timestamp, points[i].Value)
+			}
+			i++
+		}
+	}
```

Comment:
this needs a check to confirm that we actually iterated over all points (if iter.Next() didn't return points, we wouldn't notice now)